### PR TITLE
Include the namespace in the subscription durable name

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Changes proposed in this pull request:

- Include the namespace in the subscription durable name
- Add a test to check the fix

**Related issue(s)**
#1154

### CODE REVIEW

File: components/event-bus/test/integration/publish_push_test.go [hunk 1]
```diff
@@ -279,3 +279,68 @@ func Test_pushRequestShouldNotIncludeKymaTopicHeader(t *testing.T) {
...
 }
+
+func Test_sameSubjectSubscribersInDifferentNamespacesShouldReceiveEventsOfThatSubject(t *testing.T) {
+	// create two subscriptions with the same name in two different namespaces to the same subject
+	name := "test-subscription"
+	subscription1 := util.NewSubscription(name, "namespace-1", subscriberServerV1.URL+util.SubServer1EventsPath, eventType, eventTypeVersion, sourceIDV1)
+	subscription2 := util.NewSubscription(name, "namespace-2", subscriberServerV2.URL+util.SubServer2EventsPath, eventType, eventTypeVersion, sourceIDV1)
+
+	// handle the two subscriptions
+	pushApplication := initPushApplication()
```
Comment (abbi-gaurav) [inline components/event-bus/test/integration/publish_push_test.go]: All the initialization should be done in the `TestMain` including starting the applications.

Please also fix `Test_pushRequestShouldNotIncludeKymaTopicHeader` which has similiar issues

**Comment by marcobebway** [comment]
Verified that the tracing flow is working correctly, check the screenshot below.

subscription-per-env

File: components/event-bus/test/integration/publish_push_test.go [hunk 2]
```diff
@@ -279,3 +278,58 @@ func Test_pushRequestShouldNotIncludeKymaTopicHeader(t *testing.T) {
 		t.Fatalf("request to endpoint should not include the header %s", headerKymaTopic)
 	}
 }
+
+func Test_sameSubjectSubscribersInDifferentNamespacesShouldReceiveEventsOfThatSubject(t *testing.T) {
```
Comment (Abd4llA) [inline components/event-bus/test/integration/publish_push_test.go]: I think this test is indecisive. The test case we need to verify is the isolation of two subscriptions with the same name in two different namespaces. So either send an event to sub1 and verify that sub 2 didn't receive it, or send two different events to the two subscriptions and verify each received the correct one.

The current test will not fail if the bug was not fixed so it does not expose it.

Comment (Abd4llA) [reply components/event-bus/test/integration/publish_push_test.go]: You can keep the current test, but the crucial one is the isolation test IMHO.

Comment (marcobebway) [reply components/event-bus/test/integration/publish_push_test.go]: As per our offline discussion with @Abd4llA, this test is meant to verify that two subscriptions with the same name but in different kubernetes namespaces, can both receive the same events when subscribing to the same subject.

As agreed with @Abd4llA, no more action required regarding this and it is resolved.

**Pull request merged**